packages/server-renderer/__tests__/renderToString.spec.ts

```diff
@@ -236,6 +279,35 @@ describe('ssr: renderToString', () => {
       )
     })
 
+    test('nested components with template slots', async () => {
+      const Child = {
+        props: ['msg'],
+        ssrRender(ctx: any, push: any, parent: any) {
```

Comment:
this could be a template as well.

Reply:
We need another test case similar to this, with the child component using a normal render function to test compatibility between **compiled parent** + **manual render fn child**:

``` js
render() {
  return h('div', this.$slots.default({ msg: 'from slot' }))
}
```